### PR TITLE
Download data fix for new abacus structure.

### PR DESCRIPTION
**Pull request by jonathansberry**
https://github.com/meerkat-code/meerkat_issues/issues/242

### CODE REVIEW

File: api_background/api_background/export_data.py [hunk 1]
```diff
@@ -130,6 +125,23 @@ def export_category(uuid, form_name, category, download_name,
...
+    translation_dir = country_config.get("translation_dir", None)
+    if translation_dir:
+        try:
+            t = gettext.translation(
+                'messages',
+                translation_dir,
+                languages=["en", "fr"]
+            )
+        except FileNotFoundError:
+            print("Translations not found")
```
Comment (tomeksabala) [inline api_background/api_background/export_data.py]: Use logging here. And include the stack trace with `exec_info=True`.

File: api_background/api_background/export_data.py [hunk 2]
```diff
@@ -797,15 +823,19 @@ def __save_form_data(xls_csv_writer, query_form_data, operation_status, keys, al
...
+                          "data is not of a dictionary type", i)
             continue
         # Initialise empty result for header line
         row = []
         for key in keys:
             try:
                 row.append(result.data.get(key, ''))
             except AttributeError:
-                logging.exception("Error while parsing row %s with data:\n%s", result, result.data, exc_info=True)
+                logging.exception(
```
Comment (tomeksabala) [inline api_background/api_background/export_data.py]: This line break doesn't look very good here.

File: api_background/api_background/export_data.py [hunk 3]
```diff
@@ -797,15 +823,19 @@ def __save_form_data(xls_csv_writer, query_form_data, operation_status, keys, al
             logging.error("Skipping result %d. Data is None", i)
             continue
         if not isinstance(result.data, dict):
-            logging.error("Skipping result %d which data is not of a dictionary type", i)
+            logging.error("Skipping result %d which "
```
Comment (tomeksabala) [inline api_background/api_background/export_data.py]: This line break doesn't look very good here.

File: api_background/api_background/export_data.py [hunk 4]
```diff
@@ -772,21 +798,21 @@ def update_operation_status(self, status):
...
-
     def submit_operation_failure(self):
         self.download_data_file.status = 1.0
         self.download_data_file.success = 0
         self.session.commit()
 
 
-def __save_form_data(xls_csv_writer, query_form_data, operation_status, keys, allowed_location, location_data):
-    (locations, locs_by_deviceid, zones, regions, districts, devices) = location_data
+def __save_form_data(xls_csv_writer, query_form_data,
```
Comment (tomeksabala) [inline api_background/api_background/export_data.py]: This line break doesn't look very good here.

File: api_background/api_background/export_data.py [hunk 5]
```diff
@@ -733,37 +755,41 @@ def export_form(uuid, form, allowed_location, fields=None):
...
+        keys = __get_keys_from_db(db, form, param_config)
 
     xls_csv_writer = XlsCsvFileWriter(base_folder, form, uuid)
     xls_csv_writer.write_xls_row(keys)
     xls_csv_writer.write_csv_row(keys)
 
-    query_form_data = session.query(form_tables()[form].data)
-    __save_form_data(xls_csv_writer, query_form_data, operation_status, keys, allowed_location, location_data)
+    query_form_data = session.query(form_tables(param_config)[form].data)
+    __save_form_data(xls_csv_writer, query_form_data,
```
Comment (tomeksabala) [inline api_background/api_background/export_data.py]: This line break doesn't look very good here.

File: api_background/api_background/export_data.py [hunk 6]
```diff
@@ -648,7 +664,8 @@ def export_data_table(uuid, download_name,
         return_keys.append(v[1])
 
     result = session.query(*columns).filter(
-        Data.variables.has_key(restrict_by)).group_by(*groups)
+        Data.variables.has_key(restrict_by)
```
Comment (tomeksabala) [inline api_background/api_background/export_data.py]: This would look better in single line IMO.

File: meerkat_api/resources/export_data.py [hunk 7]
```diff
@@ -57,7 +57,11 @@ class ExportData(Resource):
 
     def get(self, use_loc_ids=False):
         uid = str(uuid.uuid4())
-        export_data.delay(uid, g.allowed_location, use_loc_ids)
+        yaml_config = yaml.dump(abacus_config)
+        export_data.delay(
+            uid, g.allowed_location, use_loc_ids,
+            param_config_yaml=yaml_config
+        )
```
Comment (tomeksabala) [inline meerkat_api/resources/export_data.py]: How about having args first and kwargs later with the following bracketing:
```
export_data.delay(uid, g.allowed_location, use_loc_ids,
    param_config_yaml=yaml_config)
```
And simillar in line 94.

File: meerkat_api/resources/export_data.py [hunk 8]
```diff
@@ -117,12 +123,13 @@ def get(self, form_name, category, download_name, data_type=None):
...
             return "No variables"
         language = request.args.get("language", "en")
+        yaml_config = yaml.dump(abacus_config)
         export_category.delay(uid, form_name, category,
                               download_name, variables, data_type,
                               g.allowed_location,
                               start_date=request.args.get("start_date", None),
                               end_date=request.args.get("end_date", None),
-                              language=language)
+                              language=language, param_config_yaml=yaml_config)
```
Comment (tomeksabala) [inline meerkat_api/resources/export_data.py]: `param_config_yaml` kwarg in separate line.

File: meerkat_api/resources/export_data.py [hunk 9]
```diff
@@ -229,7 +236,11 @@ def get(self, form):
             fields = request.args["fields"].split(",")
         else:
             fields = None
-        export_form.delay(uid, form, g.allowed_location, fields)
+        yaml_config = yaml.dump(abacus_config)
+        export_form.delay(
```
Comment (tomeksabala) [inline meerkat_api/resources/export_data.py]: Similar to comment in line 64.

File: api_background/api_background/export_data.py (hunk 6 above)
Comment (jonathansberry) [reply api_background/api_background/export_data.py]: PEP line lengths are there for a reason?  Not sure it is worth my time undoing code to become PEP non-compliant.

File: api_background/api_background/export_data.py (hunk 5 above)
Comment (jonathansberry) [reply api_background/api_background/export_data.py]: PEP line lengths are there for a reason?  Not sure it is worth my time undoing code to become PEP non-compliant.

File: api_background/api_background/export_data.py (hunk 4 above)
Comment (jonathansberry) [reply api_background/api_background/export_data.py]: PEP line lengths are there for a reason?  Not sure it is worth my time undoing code to become PEP non-compliant.

File: api_background/api_background/export_data.py (hunk 3 above)
Comment (jonathansberry) [reply api_background/api_background/export_data.py]: PEP line lengths are there for a reason?  Not sure it is worth my time undoing code to become PEP non-compliant.

File: api_background/api_background/export_data.py (hunk 2 above)
Comment (jonathansberry) [reply api_background/api_background/export_data.py]: PEP line lengths are there for a reason?  Not sure it is worth my time undoing code to become PEP non-compliant.

File: meerkat_api/resources/export_data.py (hunk 9 above)
Comment (jonathansberry) [reply meerkat_api/resources/export_data.py]: I don't see a comment in line 64?

File: meerkat_api/resources/export_data.py (hunk 7 above)
Comment (jonathansberry) [reply meerkat_api/resources/export_data.py]: Not sure what you are saying here. Are you just referring to indentation/style? Or something deeper?  Personally I always choose opening and closing brackets on same indentation depth - think it is far clearer and neater even if it does take up an extra line or two.  But up to you as maintainer.

File: api_background/api_background/export_data.py (hunk 6 above)
Comment (tomeksabala) [reply api_background/api_background/export_data.py]: I don't think we have agreed on any fixed line size. All of pointed out line breaks in this PR makes code only less readable and inconsistent with the rest of the file.
There are some changes in this PR that only breaks lines in the code without any other changes. I'd insist to keep the code in single line in places I've mentioned earlier.

File: meerkat_api/resources/export_data.py (hunk 7 above)
Comment (tomeksabala) [reply meerkat_api/resources/export_data.py]: I suggest here to be consistent with other function calls in this module. E.g. 
`export_category.delay` call in line 127. There opening and closing parenthesis are without line breaks.
Second thing was to keep keyword arguments in separate lines each as it's done in the delay call in line 127 too.

File: meerkat_api/resources/export_data.py (hunk 9 above)
Comment (tomeksabala) [reply meerkat_api/resources/export_data.py]: I mean my github comment I've submitted for line 64 of export_data.py. Sorry.

**Comment by jonathansberry** [comment]
This should be working although there are some outstanding stylistic issues to be addressed, and more significantly for some very strange reason Travis just broke between the last two commits with a seemingly completely unrelated "AFRO report" issue. 

@tomeksabala @Gulfa If either of you are making time up some tonight, maybe you could get this merged into master and get a deployment of Jordan going for review tomorrow morning - I'm aware it is a bit of priority. Otherwise I'm sure I can finish it up tomorrow with Gunnar in Leeds.

**Pull request merged**